src/pointer-analysis/value_set_fi.cpp

```diff
@@ -33,13 +33,13 @@ hash_numbering<irep_idt, irep_id_hash> value_set_fit::function_numbering;
 static const char *alloc_adapter_prefix="alloc_adaptor::";
 
 #define forall_objects(it, map) \
-  for(object_map_dt::const_iterator (it) = (map).begin(); \
-  (it)!=(map).end(); \
+  for(object_map_dt::const_iterator it = (map).begin(); \
+  it!=(map).end(); \
   (it)++)
```

Comment:
I think this is a good reason to get rid of those macros (just the ones here, not all `forall_*`/`Forall_*`). The use of `auto` or ranged-for loop make them unnecessary.

Reply:
Not entirely, see e.g. `pointer-analysis/value_set_fivr.cpp:202`:
```
      object_map_dt::const_iterator next(o_it);
      next++;

      if(next!=object_map.read().end())
      {
        out << "\n      ";
      }
```
I cannot construct an iterator from the value ranged over by `for(auto const &...)`, do you have a suggestion on how to deal with this case?

Reply:
But using `for(auto o_it = bla.begin(); o_it != bla.end(); ++o_it)` should still work, shouldn't it?

Reply:
I think your reviews are conflicting: I can use a ranged-for loop when I can, but where I cannot, shall I keep the macro (without the parentheses) or write the loop header explicitly with `auto`?

Reply:
I'm in favour of removing this macro (because it's broken).

Reply:
The macros are pre-ranged-for, and should likely disappear.